toolz/itertoolz.py

```diff
@@ -496,6 +496,34 @@ def concatv(*seqs):
     return concat(seqs)
 
 
+def duplicate(seq, n=1):
```

Comment:
Yeah, I was struggling for a name when I wrote this. `repeat_each` seems reasonable.